### ISSUE TITLE
echoCancelation typo in lib.d.ts, lib.dom.d.ts, lib.es6.d.ts

### ISSUE THREAD

**Issue by mkrymski** [microsoft/TypeScript#15175]
**TypeScript Version:**  2.2.1 / nightly (2.2.0-dev.201xxxxx)

**Code**

```ts
 private mediaConstraints: MediaStreamConstraints = {
    audio: {
      echoCancellation: true
    }
}

```

**Expected behavior:**
Should be compiling correctly

**Actual behavior:**

I'm having following error:

error TS2322: Type '{ audio: { echoCancellation: boolean; }; }' is not assignable to type 'MediaStreamConstraints'.
  Types of property 'audio' are incompatible.
    Type '{ echoCancellation: boolean; }' is not assignable to type 'boolean | MediaTrackConstraints'.
      Object literal may only specify known properties, and 'echoCancellation' does not exist in type 'boolean | MediaTrackConstraints'.

There is typo in MediaTrackConstraintSet.
 **echoCancelation**?: boolean | ConstrainBooleanParameters;

should be replaced to 
 **echoCancellation**?: boolean | ConstrainBooleanParameters;

**Comment by mhegazy** [comment]
PRs welcomed. You can find more information about contributing lib.d.ts fixes at https://github.com/Microsoft/TypeScript/blob/master/CONTRIBUTING.md#contributing-libdts-fixes.

**Comment by iStefo** [comment]
+1, would like to create a PR for this but I'm not sure how soon I will find time to do so

**Comment by NaridaL** [comment]
Note that this would be fixed by https://github.com/Microsoft/TSJS-lib-generator/pull/227

**Comment by saschanaz** [comment]
Seems this has been fixed.

**Comment by jakebailey** [comment]
Fixes in #22437
